src/hyperlight_host/src/hypervisor/kvm.rs

```diff
@@ -579,78 +579,31 @@ impl Hypervisor for KVMDriver {
-                crate::debug!("KVM MMIO Read -Details: Address: {} \n {:#?}", addr, &self);
-
-                match get_memory_access_violation(
-                    addr as usize,
-                    self.sandbox_regions
-                        .iter()
-                        .chain(self.mmap_regions.iter().map(|(r, _)| r)),
-                    MemoryRegionFlags::READ,
-                ) {
-                    Some(access_violation_exit) => access_violation_exit,
-                    None => HyperlightExit::Mmio(addr),
-                }
-            }
-            Ok(VcpuExit::MmioWrite(addr, _)) => {
-                crate::debug!("KVM MMIO Write -Details: Address: {} \n {:#?}", addr, &self);
-
-                match get_memory_access_violation(
-                    addr as usize,
-                    self.sandbox_regions
-                        .iter()
-                        .chain(self.mmap_regions.iter().map(|(r, _)| r)),
-                    MemoryRegionFlags::WRITE,
-                ) {
-                    Some(access_violation_exit) => access_violation_exit,
-                    None => HyperlightExit::Mmio(addr),
-                }
-            }
+    fn run_vcpu(&mut self) -> Result<HyperlightExit> {
+        match self.vcpu_fd.run() {
+            Ok(VcpuExit::Hlt) => Ok(HyperlightExit::Halt()),
+            Ok(VcpuExit::IoOut(port, data)) => Ok(HyperlightExit::IoOut(port, data.to_vec())),
+            Ok(VcpuExit::MmioRead(addr, _)) => Ok(HyperlightExit::MmioRead(addr)),
+            Ok(VcpuExit::MmioWrite(addr, _)) => Ok(HyperlightExit::MmioWrite(addr)),
             #[cfg(gdb)]
-            // KVM provides architecture specific information about the vCPU state when exiting
-            Ok(VcpuExit::Debug(debug_exit)) => match self.get_stop_reason(debug_exit) {
-                Ok(reason) => HyperlightExit::Debug(reason),
-                Err(e) => {
-                    log_then_return!("Error getting stop reason: {:?}", e);
-                }
-            },
+            Ok(VcpuExit::Debug(debug_exit)) => Ok(HyperlightExit::Debug {
+                dr6: debug_exit.dr6,
+                exception: debug_exit.exception,
+            }),
             Err(e) => match e.errno() {
-                // We send a signal (SIGRTMIN+offset) to interrupt the vcpu, which causes EINTR
-                libc::EINTR => HyperlightExit::Cancelled(),
-                libc::EAGAIN => HyperlightExit::Retry(),
-                _ => {
-                    crate::debug!("KVM Error -Details: Address: {} \n {:#?}", e, &self);
-                    log_then_return!("Error running VCPU {:?}", e);
-                }
+                // InterruptHandle::kill() sends a signal (SIGRTMIN+offset) to interrupt the vcpu, which causes EINTR
+                libc::EINTR => Ok(HyperlightExit::Cancelled()),
+                libc::EAGAIN => Ok(HyperlightExit::Retry()),
+                _ => Ok(HyperlightExit::Unknown(format!(
```

Comment:
Now this has changed to return an OK(unknown) when we get an unexpected error (which looks a bit strange) does this function need to return a Result since it looks like all returns are OK().

Seems like the other 2 implementations have limited cases where they return Err()